### ISSUE TITLE
Can this be integrated with MLJ?

### ISSUE THREAD

**Issue by PyDataBlog** [invenia/FeatureTransforms.jl#91]
It would be really neat if this idea was integrated into MLJ ecosystem in some form so Julia can have something like what Python has with scikit-learn. Custom transformers supported by the ML ecosystem. Is this possible?

**Comment by xiaodaigh** [comment]
hope not. MLJ is pretty heavy. need a lighter alternative

**Comment by PyDataBlog** [comment]
> hope not. MLJ is pretty heavy. need a lighter alternative

Which part do you want stripped?

**Comment by xiaodaigh** [comment]
with MLJ, I need to make a machine before I can do OneHotEncoding. So It's too much boiler-platy stuff for my liking

**Comment by PyDataBlog** [comment]
> with MLJ, I need to make a machine before I can do OneHotEncoding. So It's too much boiler-platy stuff for my liking

ML is rarely done in some silo but mostly some end to end pipeline. Who would want to use OHE on its own from an ML library? So it makes sense to tie everything together, I think.

**Comment by xiaodaigh** [comment]
I see. you are meant to pipeline with a machine

**Comment by PyDataBlog** [comment]
> I see. you are meant to pipeline with a machine

Exactly! So that design logic flows through most ML libraries now. This issue would complete MLJ as a fully fledged end to end ML library. Imagine shipping one binary pipeline object like everyone does with Scikit-Learn to a production system instead of multiple objects from feature engineering steps.

**Comment by rofinn** [comment]
FWIW, it looks like we could support that entire ecosystem by:

1. Depending only on MLJModelInterface.jl which is a pretty minimal package
2. Putting the MLJModelInterface.Unsupervised wrappers in a separate submodule

**Comment by PyDataBlog** [comment]
> FWIW, it looks like we could support that entire ecosystem by:
> 
> 
> 
> 1. Depending only on MLJModelInterface.jl which is a pretty minimal package
> 
> 2. Putting the MLJModelInterface.Unsupervised wrappers in a separate submodule
> 
> 
> 
> I don't think we'd need to change anything else about how our package works.

That would be awesome and significantly improve the Julia ML ecosystem!

**Comment by juliohm** [comment]
Just pointing out in case someone missed it... We addressed a couple of design issues in this package on a fresh new package called TableTransforms.jl, which supports composible, revertible pipelines: https://github.com/JuliaML/TableTransforms.jl
